src/node_state.rs

```diff
@@ -321,49 +321,68 @@ impl NodeState {
 
         let mut should_try_and_settle = true;
 
-        let blobs: Vec<UnsettledBlobMetadata> = tx
-            .blobs
-            .iter()
-            .enumerate()
-            .map(|(index, blob)| {
-                tracing::trace!("Handling blob - {:?}", blob);
-                if let Some(Ok(verifier)) = self
-                    .contracts
-                    .get(&blob.contract_name)
-                    .map(|b| TryInto::<NativeVerifiers>::try_into(&b.verifier))
-                {
-                    let hyle_output = verifiers::verify_native(
-                        blob_tx_hash.clone(),
-                        BlobIndex(index),
-                        &tx.blobs,
-                        verifier,
-                    );
-                    tracing::trace!("Nativer verifier in blob tx - {:?}", hyle_output);
-                    return UnsettledBlobMetadata {
-                        blob: blob.clone(),
-                        possible_proofs: vec![(verifier.into(), hyle_output)],
-                    };
-                } else if blob.contract_name.0 == "hyle" {
-                    // 'hyle' is a special case -> See settlement logic.
-                } else {
-                    should_try_and_settle = false;
+        let mut blobs = Vec::new();
+        for (index, blob) in tx.blobs.iter().enumerate() {
```

Comment:
why not keeping the .map as before? Is there a reason of using a for loop instead?

Reply:
one less indentation level

Reply:
Will revert here tbh, there's a few instances in this file where the indent level ends up making everything more readable IMO but maybe not here.